src/server/aws_lambda.ts

```diff
@@ -0,0 +1,56 @@
+// eslint-disable-next-line import/no-unresolved
+import { APIGatewayProxyHandler, APIGatewayProxyEvent, APIGatewayProxyResult, APIGatewayProxyCallback, Context } from 'aws-lambda'
+
+type SyncHandler = (
+    event: APIGatewayProxyEvent,
+    context: Context,
+    callback: APIGatewayProxyCallback,
+) => void;
+
+export type AsyncHandler = (
+    event: APIGatewayProxyEvent,
+    context: Context,
+) => Promise<APIGatewayProxyResult>;
+
+/**
+ * async (event) => async handler
+ * async (event, context) => async handler
+ * (event, context, callback) => sync handler
+ * @param handler
+ */
+function getAsyncHandler(handler: APIGatewayProxyHandler): AsyncHandler {
+    if (handler.length > 2) {
+        return ((event, context) => {
+                return new Promise((resolve, reject) => {
```

Comment:
I like this approach, but I wonder if they're identical? For instance, [`context.callbackWaitsForEmptyEventLoop`](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html)—if a user sets it in a sync handler, will AWS still respect it, considering we aren't using the callback anymore?

Reply:
> I like this approach, but I wonder if they're identical?

Can you please elaborate? I'm not sure I understand. The goal is to always return an async handler.

> will AWS still respect it, considering we aren't using the callback anymore?

That's a good question. My guess is that AWS will ignore it since that property only has effect in non-async handlers.

Reply:
This won't be a problem anymore since we are returning the same type of handler we receive.